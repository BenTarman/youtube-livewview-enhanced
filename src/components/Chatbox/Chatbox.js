import React from 'react'
import styled, { css } from 'styled-components'
import { rgba } from 'polished'
import PropTypes from 'prop-types'
import ChatList from '../ChatList/ChatList'

import Theme from '../../Theme'
import ChatboxSettings from '../ChatboxSettings/ChatboxSettings'

const Box = styled.div`
  background-color: ${props => rgba(props.theme.fullscreen.containerColor, props.theme.fullscreen.opacity)};
  padding: 20px;
  height: ${props => props.height}px;
  width: ${props => props.width}px;

  margin: ${props => props.top} 0 0 ${props => props.left};
`

const ToolTip = styled.div`
  background-color: green;
  margin-left: 20px;
`

// TODO: replace this with gear svg when in the mood.
const Button = styled.button`
  cursor: pointer;
  display: none;
`

const Container = styled.div`
  display: flex;

  &:hover {
    Button {
      display: inherit;
    }
  }
`

// eslint-disable-next-line react/prop-types
const Chatbox = ({ isDragging, width, height, isResizing, left, top }) => {
  return (
    <Container>
      <Box height={height} width={width} left={left} top={top}>
        <Button type="button">Edit Config (temp button)</Button>
        {!isDragging && !isResizing ? <ChatList /> : <div>You are Dragging</div>}
      </Box>

      <ToolTip>
        <ChatboxSettings />
      </ToolTip>
    </Container>
  )
}

Chatbox.propTypes = {
  isDragging: PropTypes.bool.isRequired
}

export default Chatbox
