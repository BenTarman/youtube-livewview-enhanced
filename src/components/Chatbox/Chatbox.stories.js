import React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Chatbox from './Chatbox'

const mockStore = configureStore()
const store = mockStore({
  livechat: {
    chatMessages: [
      {
        authorPhotoNode: {
          src: 'https://yt3.ggpht.com/ytc/AAUvwngn_ZtDfYYF_LlERXSOsMxnvIEGHxM22iC08w=s32-c-k-c0x00ffffff-no-rj'
        },
        authorName: 'Author',
        chatBadge: null,
        chatMessage:
          '[EN]<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u1f4ac.svg">Lorem ipsum dolor sit amet, consectetur adipiscing elit <img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u2764.svg">'
      },
      {
        authorPhotoNode: {
          src: 'https://yt3.ggpht.com/ytc/AAUvwngn_ZtDfYYF_LlERXSOsMxnvIEGHxM22iC08w=s32-c-k-c0x00ffffff-no-rj'
        },
        authorName: 'Author',
        chatBadge: null,
        chatMessage:
          '[EN]<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u1f4ac.svg">Lorem ipsum dolor sit amet, consectetur adipiscing elit <img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u2764.svg">'
      },
      {
        authorPhotoNode: {
          src: 'https://yt3.ggpht.com/ytc/AAUvwngn_ZtDfYYF_LlERXSOsMxnvIEGHxM22iC08w=s32-c-k-c0x00ffffff-no-rj'
        },
        authorName: 'Author',
        chatBadge: null,
        chatMessage:
          '[EN]<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u1f4ac.svg">Lorem<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u2764.svg">'
      },
      {
        authorPhotoNode: {
          src: 'https://yt3.ggpht.com/ytc/AAUvwngn_ZtDfYYF_LlERXSOsMxnvIEGHxM22iC08w=s32-c-k-c0x00ffffff-no-rj'
        },
        authorName: 'Author',
        chatBadge: null,
        chatMessage:
          '[EN]<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u1f4ac.svg">lorem<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u2764.svg">'
      }
    ]
  }
})

export default {
  title: 'Chatbox',
  component: Chatbox,
  decorators: [
    Story => (
      <Provider store={store}>
        <Story />
      </Provider>
    )
  ]
}

// Putting random haachama stream in background as picture to help better simulate the ui.

export const FullscreenLiveviewOverlay = () => (
  <Chatbox isDragging={false} width={500} height={500} isResizing={false} left={0} top={0} />
)
