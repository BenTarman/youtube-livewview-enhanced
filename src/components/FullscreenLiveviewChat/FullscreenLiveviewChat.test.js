/* eslint-disable no-console */
/* eslint-disable no-undef */
import React from 'react'
import { wrapInTestContext } from 'react-dnd-test-utils'
import { cleanup, render } from '@testing-library/react'
import HTML5Backend from 'react-dnd-html5-backend'
import ReactTestUtils from 'react-dom/test-utils'
import { shallow, mount } from 'enzyme'
import { DragDropContext } from 'react-dnd'

import TestBackend from 'react-dnd-test-backend'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import FullscreenLiveviewchat from './FullscreenLiveviewChat'
import Chatbox from '../Chatbox/Chatbox'

const mockStore = configureStore()

describe('My Connected React-Redux Component', () => {
  let store

  /* TODO: So when testing in Jest it seems it thinks the styled component is a functional component
  and therefore cannot give it a ref. Will just disable the warning for this local test file for now,
  probably a bug with jest or styled components library */
  const originalWarn = console.error.bind(console.error)
  beforeAll(() => {
    console.error = msg => !msg.toString().includes('Function') && originalWarn(msg)
  })
  afterAll(() => {
    console.error = originalWarn
  })

  beforeEach(() => {
    store = mockStore({
      draggableAndResizableBox: {
        a: { top: 500, left: 500, height: 200, width: 200, isResizing: true }
      },

      livechat: {
        chatMessages: [
          {
            authorPhotoNode: {
              src: 'https://yt3.ggpht.com/ytc/AAUvwngn_ZtDfYYF_LlERXSOsMxnvIEGHxM22iC08w=s32-c-k-c0x00ffffff-no-rj'
            },
            authorName: 'Author',
            chatBadge: null,
            chatMessage:
              '[EN]<img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u1f4ac.svg">Lorem ipsum dolor sit amet, consectetur adipiscing elit <img src="https://www.youtube.com/s/gaming/emoji/9cda877f/emoji_u2764.svg">'
          }
        ]
      }
    })
  })

  // Note: not using enzyme here since doesn't work well with jest snapshots and mocking the redux store.
  it('Should render and have correct styling and layout from last snapshot', () => {
    const FullscreenLiveviewchatContext = wrapInTestContext(FullscreenLiveviewchat)
    const { asFragment } = render(
      <Provider store={store}>
        <FullscreenLiveviewchatContext />
      </Provider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  /*
  it('Should render and have correct styling and layout from last snapshot', () => {
    const FullscreenLiveviewchatContext = wrapInTestContext(FullscreenLiveviewchat)

    const componentInstance = ReactTestUtils.renderIntoDocument(
      <Provider store={store}>
        <FullscreenLiveviewchatContext />
      </Provider>
    )


    // const backend = root.getManager().getBackend()


    const piece = ReactTestUtils.findRenderedComponentWithType(root, Piece)
    backend.simulateBeginDrag([piece.getHandlerId()])
    expect(piece.render().props.isDragging).toBeTruthy()

  })
      */
})
