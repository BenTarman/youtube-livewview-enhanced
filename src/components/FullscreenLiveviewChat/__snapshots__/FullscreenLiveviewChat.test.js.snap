// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component Should render and have correct styling and layout from last snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-fubCfw dqMxbi"
  >
    <div
      class="sc-iBPRYJ dlAidR react-resizable"
      transform="translate3d(500px, 500px, 0)"
    >
      asdfasfd
      <div
        class="sc-jSgupP cjTDZD"
        height="200"
        width="200"
      >
        <div>
          You are Dragging
        </div>
      </div>
      <span
        class="react-resizable-handle react-resizable-handle-sw"
      />
      <span
        class="react-resizable-handle react-resizable-handle-se"
      />
      <span
        class="react-resizable-handle react-resizable-handle-nw"
      />
      <span
        class="react-resizable-handle react-resizable-handle-ne"
      />
      <span
        class="react-resizable-handle react-resizable-handle-w"
      />
      <span
        class="react-resizable-handle react-resizable-handle-e"
      />
      <span
        class="react-resizable-handle react-resizable-handle-n"
      />
      <span
        class="react-resizable-handle react-resizable-handle-s"
      />
    </div>
  </div>
</DocumentFragment>
`;
